Storage/src/Connection/Rest.php

```diff
@@ -628,6 +629,148 @@ private function sendWithRetry($resource, $method, $args)
             $args
         );
 
+        $requestHash = Uuid::uuid4()->toString();
+        $args['onRetryException'] = function (
+            \Exception $e,
+            $currentAttempt,
+            &$arguments
+        ) use ($requestHash) {
+            // Since we the the last attempt number here, so incrementing it
+            // to get the current attempt count
+            ++$currentAttempt;
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $requestHash,
+                $currentAttempt
+            );
+        };
+
+        $args['onExecutionStart'] = function (&$arguments) use ($requestHash) {
+            $this->updateRetryInovationHeaders(
+                $arguments,
+                $requestHash
+            );
+        };
+
         return $this->send($resource, $method, $args);
     }
+
+    /**
+     * Updates the x-goog-api-client header value with UUID
+     * and retry count
+     *
+     * @return void
+     */
+    private function updateRetryInovationHeaders(
+        &$arguments,
+        $requestHash,
+        $currentAttempt = 0
+    ) {
+        $valueToAdd = sprintf("gccl-invocation-id/%s", $requestHash);
+        $this->updateHeader(
+            'x-goog-api-client',
```

Comment:
I think let's use this header as a const in `RequestWrapper` and use it here instead of hard coding it?

Reply:
Yeah, this would make the value more central. Thanks for pointing out.

Implemented in commit [33ba71f](https://github.com/googleapis/google-cloud-php/pull/5741/commits/33ba71f14ba1fd93fbfff19cb498b8e566f0d540)